apps/web/screens/Organization/Syslumenn/Homestay.tsx

```diff
@@ -198,7 +201,13 @@ const Homestay: Screen<HomestayProps> = ({
   )
 }
 
-Homestay.getInitialProps = async ({ apolloClient, locale, query }) => {
+Homestay.getInitialProps = async ({ apolloClient, locale, pathname }) => {
+  const slug = pathname.split('/')?.[locale !== 'is' ? 3 : 2] ?? 'syslumenn'
+  const subSlug =
+    pathname.split('/')?.[locale !== 'is' ? 4 : 3] ?? 'heimagisting'
```

Comment:
Can you make this more robust? Maybe searching for parameters or regex? Indexing slug seems fragile.

Reply:
If we so happen to index out of the array then javascript will return undefined and the subSlug will become 'heimagisting'. So I believe that this is stable even though it looks sketchy. Perhaps moving this into a utility function would allow us to revisit this at a later point if the url structure changes in the future

Reply:
I moved this into a util function, I don't see a better way at this moment how to make this more robust, this was previously hard coded in which didn't work since the english and icelandic slugs are different.

Reply:
I think your solution is good. At least this way we only need to change the url lookup once. Is this overengineering? I like this better than the alternative 🤷